flink-connectors/flink-connector-mongodb/src/main/java/org/apache/flink/mongodb/table/MongodbDynamicTableSourceSinkFactory.java

```diff
@@ -0,0 +1,133 @@
+            .withDescription("The batch size when table invoking.");
+
+    public static final ConfigOption<String> FORMAT =
+            ConfigOptions.key("format")
+                    .stringType()
+                    .defaultValue("json")
+                    .withDescription(
+                            "Defines the format identifier for encoding data. "
+                                    + "The identifier is used to discover a suitable format factory.");
+
+    @Override
+    public DynamicTableSink createDynamicTableSink(Context context) {
+        ContextUtil.transformContext(this, context);
+        FactoryUtil.TableFactoryHelper helper = FactoryUtil.createTableFactoryHelper(this, context);
+        helper.validate();
+
+        MongodbSinkConf mongodbSinkConf = new MongodbSinkConf(
+                (String) helper
+                        .getOptions()
+                        .get(DATABASE),
+                (String) helper.getOptions().get(COLLECTION_NAME),
+                (String) helper.getOptions().get(URI),
+                ((Integer) helper.getOptions().get(MAX_CONNECTION_IDLE_TIME)).intValue(),
+                ((Integer) helper.getOptions().get(BATCH_SIZE)).intValue());
+
+        TableSchema physicalSchema = TableSchemaUtils.getPhysicalSchema(context
+                .getCatalogTable()
+                .getSchema());
+        LOG.info("Create dynamic mongoDB table table: {}.", mongodbSinkConf);
+        return new MongodbDynamicTableSink(mongodbSinkConf, physicalSchema);
+    }
+
+    @Override
+    public String factoryIdentifier() {
+        return IDENTIFIER;
+    }
+
+    @Override
+    public Set<ConfigOption<?>> requiredOptions() {
+        Set<ConfigOption<?>> requiredOptions = new HashSet();
+        requiredOptions.add(DATABASE);
+        requiredOptions.add(COLLECTION_NAME);
+        requiredOptions.add(URI);
+        return requiredOptions;
+    }
+
+    @Override
+    public Set<ConfigOption<?>> optionalOptions() {
+        Set<ConfigOption<?>> optionals = new HashSet();
+        optionals.add(MAX_CONNECTION_IDLE_TIME);
+        optionals.add(BATCH_SIZE);
+        return optionals;
+    }
+
+    @Override
+    public DynamicTableSource createDynamicTableSource(Context context) {
+        ContextUtil.transformContext(this, context);
+        FactoryUtil.TableFactoryHelper helper = FactoryUtil.createTableFactoryHelper(this, context);
+        MongodbSinkConf mongodbSinkConf = new MongodbSinkConf(
+                (String) helper
+                        .getOptions()
+                        .get(DATABASE),
+                (String) helper.getOptions().get(COLLECTION_NAME),
+                (String) helper.getOptions().get(URI),
+                ((Integer) helper.getOptions().get(MAX_CONNECTION_IDLE_TIME)).intValue(),
+                ((Integer) helper.getOptions().get(BATCH_SIZE)).intValue());
+
+        TableSchema physicalSchema = TableSchemaUtils.getPhysicalSchema(context
```

Comment:
Please migrate this from the deprecated `TableSchema` stack to the new `Schema` stack. In fact, here we don't want to pass the schema along at all, but rather only the data type for it. You can access that through `context.getPhysicalRowDataType()`.

Reply:
@Airblader Thank you for your guidance. I have started to reconstruct the Table, and I will present a version tonight

Reply:
@Airblader It's gonna be the day after tomorrow, and it's gonna be harder than I thought.

Reply:
Can you sketch out why it is difficult? Do you rely on having the actual schema somewhere? We recently introduced several utility methods also on `DataTypes` etc. to make working with this easier.

Reply:
@Airblader Schema changes have been made.
I said the difficulty I had was this:
1. I want to imitate Kafka's TableAPI and use the Mongo connector that implements the Sink interface instead of the RichSinkFunction interface in getSinkRuntimeProvider. However, I encountered a serialization problem after modification, which could not be solved.This code is not very familiar, it may take time to look, in order to locate the problem.
2. Parameters of mongo client should be added and standardized according to official documents of Mongo
3. When orC data contains complex types such as: List, parsing failure will occur after importing Mongo, which requires a mechanism to convert complex types.